torch/amp/autocast_mode.py

```diff
@@ -323,14 +324,23 @@ def __init__(
                     "Current CUDA Device does not support bfloat16. Please switch dtype to float16."
                 )
         elif self.device == "mps":
-            supported_dtype = [torch.float16]
+            supported_dtype = [torch.bfloat16, torch.float16]
             if self.fast_dtype not in supported_dtype:
-                error_message = "In MPS autocast, but the target dtype is not supported. Disabling autocast.\n"
-                error_message += (
-                    "MPS Autocast only supports dtype of torch.float16 currently."
+                error_message = (
+                    "In MPS autocast, but the target dtype is not supported. Disabling autocast.\n"
+                    "MPS Autocast only supports dtype of torch.bfloat16 and torch.float16 currently."
                 )
                 warnings.warn(error_message)
                 enabled = False
+            elif self.fast_dtype == torch.bfloat16:
+                mac_ver = platform.mac_ver()[0]
+                if mac_ver and tuple(map(int, mac_ver.split("."))) < (14, 0):
```

Comment:
This function actually checks if MPS framework supports certain functions
```suggestion
                if not torch.backends.mps.is_macos_or_newer(14, 0):
```

Reply:
Good catch! :)